authentication/ruby/oauth/microsoft_kiota_authentication_oauth/lib/microsoft_kiota_authentication_oauth/oauth_access_token_provider.rb

```diff
@@ -53,23 +51,25 @@ def get_authorization_token(uri, additional_properties = {})
 
       raise StandardError, 'Only https is supported' if parsed_url.scheme != 'https'
 
-      if @cached_token
-        token = OAuth2::AccessToken.from_hash(@token_request_context.oauth_provider, @cached_token) 
-        return token.token if !token.nil? && !token.expired?
+      Fiber.new do
+        if @cached_token
+          token = OAuth2::AccessToken.from_hash(@token_request_context.oauth_provider, @cached_token) 
+          return token.token if !token.nil? && !token.expired?
 
-        if token.expired?
-          token = token.refresh!
-          @cached_token = token.to_hash
-          return token.token
+          if token.expired?
+            token = token.refresh!
+            @cached_token = token.to_hash
+            return token.token
+          end
         end
-      end
 
-      token = nil
-      token = @token_request_context.get_token
+        token = nil
+        token = @token_request_context.get_token
 
-      @cached_token = token.to_hash unless token.nil?
-      token = token.token unless token.nil?
-      token
+        @cached_token = token.to_hash unless token.nil?
+        token = token.token unless token.nil?
```

Comment:
I think type changes with this assignment for the `token` variable, so it might be a good idea to use a different name for encapsulating `token`.

Reply:
actually the assignment is useless, updated. Thanks for the suggestion